src/ui/public/vis/vis_update_state.js

```diff
@@ -84,6 +85,9 @@ export const updateOldState = (visState) => {
     delete newState.params.metric.autoExtend;
     newState.params.metric.metricColorMode = newState.params.metric.gaugeColorMode;
     delete newState.params.metric.gaugeColorMode;
+  } else if(visState.type === 'metric' && _.get(visState.params, 'gauge.gaugeType', 'Metric') !== 'Metric') {
```

Comment:
We now assume `Metric` as default, because in case you only created a metric and never switched to gauge, `gauge.gaugeType` won't even exist. So we only want to transfer to gauge if we really have a gaugeType, that is not `Metric`.